language/move-lang/src/parser/byte_string.rs

```diff
@@ -0,0 +1,355 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::errors::*;
+use byteorder::{BigEndian, ByteOrder};
+use std::convert::TryFrom;
+
+use crate::parser::syntax::make_loc;
+
+/// Escape sequence.
+enum Sequence {
+    /// \x00
+    Hex(String),
+    ///u{7FFF}
+    Unicode(String),
+    /// \
+    Linebreak,
+    /// The start of an escape sequence. \
+    Generic,
+}
+
+#[derive(Default)]
+pub struct ByteStringCoder {
+    filename: &'static str,
+    start_offset: usize,
+    buffer: Vec<u8>,
+    escape_sequence: Option<Sequence>,
+}
+
+impl ByteStringCoder {
+    pub fn new(filename: &'static str, start_offset: usize) -> ByteStringCoder {
+        ByteStringCoder {
+            filename,
+            start_offset,
+            buffer: vec![],
+            escape_sequence: None,
+        }
+    }
+
+    pub fn decode(mut self, text: &str) -> Result<Vec<u8>, Error> {
+        for (idx, c) in text.chars().enumerate() {
+            self.next_char(idx, c)?;
+        }
+
+        if let Some(escape_sequence) = &self.escape_sequence {
```

Comment:
Can you add a comment before this to explain the context? E.G., this is checking for incomplete escape sequences at the end of the string.

I also wonder if the errors here would be more helpful if they described the situations as incomplete escapes rather than saying that there is an invalid character. Which character is invalid? The closing quote? The loc that is reported does not point at an invalid character (which is what I would expect with that message).

Reply:
Changed the error description.